src/device-registry/utils/nearest-device.js

```diff
@@ -0,0 +1,61 @@
+const { logElement } = require("./log");
+
+const nearestDevices = {
+
+    findNearestDevices: ( devices, radius, latitude, longitude ) => {
+        
+        try {
+
+            let nearest_devices = []
+
+            devices.forEach(device => {
+
+                if ('latitude' in device && 'longitude' in device && device['isActive'] == true) {
+                    distance = nearestDevices.calculateDistance(latitude, longitude, device['latitude'], device['longitude']);
+
+                    if (distance < radius) {
+                        logElement("Found a node ", distance);
+                        device['distance'] = radius;
```

Comment:
What could be the significance of this assignment? That is >> `device['distance'] = radius;`

Reply:
Thanks, @Baalmart, it's supposed to be `device['distance'] = distance`. Returning the distance from the node to the provided coordinates